File: scripts/get_bot_oauth.py
from os import environ as env

from dotenv import load_dotenv
from requests import post


def main():

    load_dotenv()
    api_key = env.get("TWITTER_API_KEY")
    str_input = lambda prompt: str(input(prompt))
    
    developer_url = f"https://twitter.com/oauth/request_token?oauth_consumer_key={api_key}&oauth_callback=oob"
    print(f"[!] Login to your *DEVELOPER* account and visit this URL:\n{developer_url}")

    oauth_token = str_input("[?] Enter the OAuth token: ")
    bot_url = f"https://twitter.com/oauth/authenticate?oauth_token={oauth_token}"
    print(f"[!] Login to your *BOT* account and visit this URL:\n{bot_url}")

    pin = str_input("[?] Enter the PIN: ")
    response = post(f"https://twitter.com/oauth/access_token?oauth_token={oauth_token}&oauth_verifier={pin}")
    
    print("[*] Your OAuth tokens are:")
    [print(token) for token in response.text.split("&")[0:2]]

if __name__ == '__main__':
    main()

File: scripts/get_bot_oauth.ts
import { config } from '@/config'
import { stdin, stdout } from 'process'
import { createInterface } from 'readline/promises'

async function main() {
  const rl = createInterface(stdin, stdout)

  const developer_url = `https://twitter.com/oauth/request_token?oauth_consumer_key=${config.TWITTER_API_KEY}&oauth_callback=oob`
  console.log(`[!] Login to your *DEVELOPER* account and visit this URL:\n${developer_url}`)

  const oauth_token = await rl.question(`[?] Enter the OAuth token: `)
  const bot_url = `https://twitter.com/oauth/authenticate?oauth_token=${oauth_token}`
  console.log(`[!] Login to your *BOT* account and visit this URL:\n${bot_url}`)

  const pin = await rl.question('[?] Enter the PIN: ')
  const response = await fetch(
    `https://twitter.com/oauth/access_token?oauth_token=${oauth_token}&oauth_verifier=${pin}`,
    { method: 'POST' }
  )

  console.log('[*] Your OAuth tokens are:')
  const oauth_tokens = await response.text()
  oauth_tokens.split('&').slice(0, 2).map(console.log)
}

main()
